src/components/Search/SearchListWithHeader.tsx

```diff
@@ -100,6 +103,7 @@ function SearchListWithHeader(
 
             if (item.transactions.every((transaction) => selectedItems[transaction.keyForList]?.isSelected)) {
                 const reducedSelectedItems: SelectedTransactions = {...selectedItems};
+                setSelectedReports([...selectedReports.filter((reportID) => reportID !== item.reportID)]);
```

Comment:
if you want, `useState` can also accept an updater function instead of the state, and perhaps in this case it would look better.

I have no opinion on this, just wanted to offer a suggestion :)
https://react.dev/reference/react/useState#setstate-parameters
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Reply:
Looks better!